purchase_triple_discount/models/product_supplierinfo.py

```diff
@@ -0,0 +1,80 @@
+# Copyright 2019 Tecnativa - David Vidal
+# Copyright 2019 Tecnativa - Pedro M. Baeza
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+import functools
+
+from odoo import api, fields, models
+
+
+class ProductSupplierInfo(models.Model):
+    _inherit = "product.supplierinfo"
+
+    discount = fields.Float(
+        compute="_compute_discount",
+        store=True,
+    )
+    discount1 = fields.Float(
+        string="Disc. (%)",
+        digits="Discount",
+        readonly=False,
+    )
+
+    discount2 = fields.Float(
+        string="Disc. 2 (%)",
+        digits="Discount",
+        readonly=False,
+    )
+    discount3 = fields.Float(
+        string="Disc. 3 (%)",
+        digits="Discount",
+        readonly=False,
+    )
+
+    _sql_constraints = [
+        (
+            "discount1_limit",
+            "CHECK (discount1 <= 100.0)",
+            "Discount 1 must be lower than 100%.",
+        ),
+        (
+            "discount2_limit",
+            "CHECK (discount2 <= 100.0)",
+            "Discount 2 must be lower than 100%.",
+        ),
+        (
+            "discount3_limit",
+            "CHECK (discount3 <= 100.0)",
+            "Discount 3 must be lower than 100%.",
+        ),
+    ]
+
+    @api.depends("discount1", "discount2", "discount3")
```

Comment:
Suggestion
@api.depends(lambda self: self._get_multiple_discount_field_names())

Reply:
I had never used it like this, I didn't know it was possible.
I have already changed it, but I have made other changes:
  - I have moved all the code that was repeated in purchase.order.line and in purchase.supplierinfo to an AbstractModel, to avoid repeating code
  - All functions that obtained the values of dicount1 , discount2, etc... use _get_multiple_discount_field_names() to obtain the fields involved